src/validate.ts

```diff
@@ -81,8 +81,10 @@ export const validate = <S extends AnySchema>(schema: S) => (value: unknown): Ty
 
     const validatorValues = Object.values(validators);
     const allValidatorKeysCount = validatorValues.length;
-    const requiredValidatorKeysCount =
-      allValidatorKeysCount - validatorValues.filter(isOptionalSchema).length;
+    const requiredValidatorKeysCount = validatorValues.reduce(
+      (acc, schema) => acc + (isOptionalSchema(schema) ? 0 : 1),
+      0,
+    );
```

Comment:
```suggestion
    const requiredValidatorKeysCount = validatorValues.reduce(
      (acc, schema) => isOptionalSchema(schema) ? ++acc : acc,
      0,
    );
```

Reply:
`Assignment to function parameter 'acc'. eslint no-param-reassign`